Source/WebCore/rendering/svg/RenderSVGContainer.cpp

```diff
@@ -112,6 +112,7 @@ void RenderSVGContainer::layoutChildren()
 FloatRect RenderSVGContainer::strokeBoundingBox() const
 {
     if (!m_strokeBoundingBox) {
+        m_strokeBoundingBox = FloatRect { };
         SVGBoundingBoxComputation boundingBoxComputation(*this);
         m_strokeBoundingBox = boundingBoxComputation.computeDecoratedBoundingBox(SVGBoundingBoxComputation::strokeBoundingBoxDecoration);
```

Comment:
Yes, this is deliberate. The reason is that our current LSBE code is not detecting recursion of this invoked from repaint bounding box computation for markers / SVG resources (like, filters, clippers, maskers). As a result, without filling it to empty stroke-bounding-box before starting computation, we will have infinite recursion in some of tests.

The example test is https://github.com/WebKit/WebKit/blob/main/LayoutTests/svg/custom/svg-reference-cycle-between-two-svg-elements.html. This is also referencing each other in a circular manner.

Right now, I cannot find the right behavior in that case from the spec. It looks like the spec does infinite recursion. Here, I assigned empty one before starting recursion so that we can stop the recursion since `m_strokeBoundingBox` is no longer `std::nullopt`. (So we do not need to crash).
Ideally we would like to have some specified behavior for that, but right now, it is just *undefined*. (Current our implementation would return the stale previous stroke-bounding-box)

The part of the issue of this is that we are using accurate painting bounding box for markers, which invokes stroke-bounding-box. So, marker -> accurate repaint bounding box -> stroke bounding box -> another marker's root's stroke bounding box -> another marker -> accurate repaint bounding box -> going to the first marker. Previously we were using non accurate repaint bounding box. Then in that case, we were using stale repaint bounding box. Either behavior looks incorrect anyway, and even spec does not answer to the right expected behavior (And it makes sense since this is repaint bounding box related issue, but I think we don't have a reasonable way to represent this recursive SVG). Thus, we are just keeping the current status-quo roughly: stopping the recursion.

Reply:
OK, can you please add a one line comment just above the first assignment, in all of the functions that do this, to explain why? Just something like:

```
// Initialize m_strokeBoundingBox before calling computeDecoratedBoundingBox, since recursively referenced markers can cause us to re-enter here.
```